File: app/src/main/java/XiYue/SiyoX/SiyoXApp.java


package XiYue.SiyoX;

import android.app.Application;
import XiYue.SiyoX.data.AppSettings;
import XiYue.SiyoX.data.VerifyManager;

public class SiyoXApp extends Application {

    private static SiyoXApp instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        AppSettings.init(this);
        VerifyManager.init(this);
    }

    public static SiyoXApp getInstance() {
        return instance;
    }
}

File: app/src/main/kotlin/XiYue/SiyoX/SiyoXApp.kt
// Copyright 2026, SiyoX contributors
// SPDX-License-Identifier: Apache-2.0

package XiYue.SiyoX

import android.app.Application
import XiYue.SiyoX.data.AppSettings
import XiYue.SiyoX.data.VerifyManager

class SiyoXApp : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        AppSettings.init(this)
        VerifyManager.init(this)
    }

    companion object {
        lateinit var instance: SiyoXApp
            private set
    }
}
